src/codeu/chat/server/Server.java

```diff
@@ -66,14 +73,44 @@
   private final Relay relay;
   private Uuid lastSeen = Uuid.NULL;
 
-  private static ServerInfo info;
 
-  public Server(final Uuid id, final Secret secret, final Relay relay) {
+  public Server(final Uuid id, final Secret secret, final Relay relay) throws IOException {
```

Comment:
Avoid throwing checked exceptions in constructors or public methods. Either

* handle the exception, or
* if it can only be caused by programmer error, throw an unchecked exception (a subclass of `RuntimeException`)